sapp/ui/frontend/src/Filter.js

```diff
@@ -187,12 +187,76 @@ const Paths = (
   );
 };
 
+const MatchesIsField = (
+  props: $ReadOnly<{
+    allOptions: mixed,
+    parameterName: string,
+    currentFilter: FilterDescription,
+    setCurrentFilter: FilterDescription => void,
+  }>,
+): React$Node => {
+  const [mode, setMode] = useState('is');
+  const [inputValue, setInputValue] = useState(null);
+
+  return (
+    <Row gutter={gutter}>
+      <Col span={6}>
+        <Select
+          options={[{value: 'is'}, {value: 'matches'}]}
+          value={mode}
+          onChange={setMode}
+          style={{width: '100%'}}
+        />
+      </Col>
+      <Col span={16}>
+        {mode === 'is' ? (
+          <Select
+            mode="multiple"
+            value={props.currentFilter[props.parameterName]}
+            options={props.allOptions.map(value => ({value}))}
+            style={{width: '100%'}}
+            onChange={value =>
+              props.setCurrentFilter({...props.currentFilter, value})
+            }
+          />
+        ): (
+          <Input
+            placeholder="regular expression"
+            style={{width: '100%'}}
+            value={inputValue}
+            onChange={event => {
+              const value = event.target.value;
+              setInputValue(value);
+              const option_values = props.allOptions.filter(option =>
+                  option.match(value),
+              );
+              let values = {};
+              values[props.parameterName] = option_values;
+              props.setCurrentFilter({...props.currentFilter, ...values});
+            }}
+            suffix={
+              <Tooltip
+                title={(props.currentFilter[props.parameterName] || []).join('\n')}
+                placement="bottom">
+                <Text type="secondary" size="small">
+                  {(props.currentFilter[props.parameterName] || []).length}
+                </Text>
+              </Tooltip>
+            }
+          />
+        )}
+      </Col>
+    </Row>
+  );
+};
+
 const SourceNames = (
   props: $ReadOnly<{
     currentFilter: FilterDescription,
     setCurrentFilter: FilterDescription => void,
   }>,
 ): React$Node => {
+
```

Comment:
Is the change here coming from the auto-formatter?

Reply:
Yes, kinda added a wierd vim plugin. Will remove it.